File: Frog Jump/Recursion.py
def minimumJump(size,arr):
    if size == 0: return 0
    if size < 0: return float('inf')

    right = left = float('inf')
    if size - 1 >= 0: left = abs(arr[size] - arr[size-1]) + minimumJump(size-1,arr)
    if size - 2 >= 0: right = abs(arr[size] - arr[size-2]) + minimumJump(size-2,arr)

    return min(left,right)

def FrogJump():
    arr = [7 ,5 ,5 ,8 ,4 ,9 ,1 ,1, 10 ]
    n = len(arr)
    return minimumJump(n-1,arr)

if __name__ == "__main__":
    print(FrogJump())

File: Frog Jump/Recursion.java

class Recursion {

    int FrogJump(int n,int[] heights) {
        if(n == 0) return 0;
        if(n < 0) return Integer.MAX_VALUE;

        int left = Integer.MAX_VALUE, right =  Integer.MAX_VALUE;
        if(n - 1 >= 0) left = Math.abs(heights[n] - heights[n-1]) + FrogJump(n-1, heights);
        if(n - 2 >= 0) right = Math.abs(heights[n] - heights[n-2]) + FrogJump(n-2, heights);

        return Math.min(left, right);
    }
    public static void main(String args[]) {
        int[] heights = {7 ,5 ,5 ,8 ,4 ,9 ,1 ,1, 10};
        int n =  heights.length;

        Recursion obj =  new Recursion();

        System.out.println(obj.FrogJump(n-1,heights));
    }
}